torch/linalg/__init__.py

```diff
@@ -235,6 +235,102 @@
     torch.return_types.linalg_slogdet(sign=tensor(-1.), logabsdet=tensor(-0.2776))
 """)
 
+eig = _add_docstr(_linalg.linalg_eig, r"""
+linalg.eig(input, *, out=None) -> (Tensor, Tensor)
+
+Computes the eigenvalues and eigenvectors of a square
+matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+
+For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
+`V` decompose the :attr:`input` such that `input = V diag(w) V⁻¹`, where `V⁻¹` is the inverse of `V`.
+
+The eigenvalues are not necessarily sorted.
+For real-valued input the resulting eigenvalues will be of complex type, unless the imaginary part is zero
+in which case it will be cast to a real type.
+
+Supports input of float, double, cfloat and cdouble dtypes.
+
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+
+.. note:: The eigenvalues and eigenvectors are computed using LAPACK's and MAGMA's `geev` routine.
+
+.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
+          a valid eigenvector. This function may compute different eigenvector representations on
+          different device types. Usually the difference is only in the sign of the eigenvector.
+
+.. note:: See :func:`torch.linalg.eigvals` for a related function that computes only eigenvalues.
+          However, that function is not differentiable.
```

Comment:
But isn't this function also not differentiable? The docs should probably reflect that, even if the state is only temporary

Reply:
Added a warning about that.